src/test/java/io/cdap/plugin/couchbase/etl/CouchbaseSourceETLTest.java

```diff
@@ -220,23 +348,62 @@ private JsonDocument getTestDocumentByCreationTime(long created) {
       .orElse(null);
   }
 
-  private void assertEquals(JsonDocument expected, StructuredRecord actual) {
+  private void assertNumericEquals(JsonDocument expected, StructuredRecord actual) {
     JsonObject content = expected.content();
-    Assert.assertEquals(content.getBoolean("boolean"), actual.<Boolean>get("boolean"));
     Assert.assertEquals(content.getDouble("number_double"), actual.<Double>get("number_double"));
     Assert.assertEquals(content.getInt("number_int"), actual.<Integer>get("number_int"));
     Assert.assertEquals(content.getLong("number_long"), actual.<Long>get("number_long"));
     Assert.assertEquals(content.getBigDecimal("number_decimal"), actual.getDecimal("number_decimal"));
     Assert.assertEquals(content.getBigInteger("number_big_int"), actual.getDecimal("number_big_int").unscaledValue());
-    Assert.assertEquals(content.getString("string"), actual.<String>get("string"));
 
-    JsonObject innerObjectExpected = content.getObject("object");
-    StructuredRecord innerObjectActual = actual.get("object");
-    Assert.assertEquals(innerObjectExpected.getString("key"), innerObjectActual.<String>get("key"));
+    JsonArray numberArrayExpected = content.getArray("number_array");
+    List<Number> numberArrayActual = actual.get("number_array");
+    Assert.assertEquals(numberArrayExpected.toList(), numberArrayActual);
+  }
+
+  private void assertNumericEqualsAsStrings(JsonDocument expected, StructuredRecord actual) {
+    JsonObject content = expected.content();
+    Assert.assertEquals(content.getDouble("number_double").toString(), actual.<String>get("number_double"));
+    Assert.assertEquals(content.getInt("number_int").toString(), actual.<String>get("number_int"));
+    Assert.assertEquals(content.getLong("number_long").toString(), actual.<String>get("number_long"));
+
+    if (new BigDecimal("0.00").equals(content.getBigDecimal("number_decimal"))) {
```

Comment:
it should not be any conditional expressions in tests

Reply:
Conditional expression removed.